stonesoup/detector/beamformers.py

```diff
@@ -0,0 +1,505 @@
+
+        for t in winstarts:
+            # calculate covariance estimate
+            R = np.matmul(np.transpose(y[t:t+windowm1]), y[t:t+windowm1])
+            R_inv = np.linalg.inv(R)
+
+            maxF = 0
+            maxtheta = 0
+
+            for theta in thetavals:
+                for phi in phivals:
+                    # convert from spherical polar coordinates to cartesian
+                    a[0] = math.cos(theta)*math.sin(phi)
+                    a[1] = math.sin(theta)*math.sin(phi)
+                    a[2] = math.cos(phi)
+                    a = a/math.sqrt(np.sum(a*a))
+                    for n in range(0, N):
+                        phase = np.sum(a*np.transpose(z[n, ]))/c
+                        v[n] = math.cos(phase) - math.sin(phase)*1j
+                    F = 1/((window-N)*np.transpose(np.conj(v))@R_inv@v)
+                    if F > maxF:
+                        maxF = F
+                        maxtheta = theta
+                        maxphi = phi
+
+            # Defining a detection
+            state_vector = StateVector([maxtheta, maxphi])  # [Azimuth, Elevation]
+            covar = CovarianceMatrix(np.array([[1, 0], [0, 1]]))
+            measurement_model = LinearGaussian(ndim_state=4, mapping=[0, 2],
+                                               noise_covar=covar)
+            current_time = current_time + timedelta(milliseconds=1000*window/self.fs)
+            detection = Detection(state_vector, timestamp=current_time,
+                                  measurement_model=measurement_model)
+            detections = set([detection])
+
+            scans.append((current_time, detections))
+
+        # For every timestep
+        for scan in scans:
+            yield scan[0], scan[1]
+
+
+class rjmcmc(Base, BufferedGenerator):
+    """A parameter estimation algorithm for a sensor array measuring passive signals. Given the
+    input signals from the array, the algorithm uses reversible-jump Markov chain Monte Carlo [1]
+    to sample from the posterior probability for a model where the number of targets and directions
+    of arrival are the unknown parameters. The algorithm is based on the work in [2] and [3].
+
+    [1] P. J. Green, Reversible jump Markov chain Monte Carlo computation and Bayesian
+    model determination, Biometrika 82(4):711-732 (1995)
+    [2] C. Andrieu and A. Doucet, Joint Bayesian Model Selection and Estimation of
+    Noisy Sinusoids via Reversible Jump MCMC, IEEE Trans. Signal Process. 47(10):2667-2676 (1999)
+    [3] C. Andrieu, N. de Freitas and A. Doucet, Robust Full Bayesian Learning for Radial Basis
+    Networks, Neural Computation 13:2359–2407 (2001)
+
+    """
+    csv_path: str = Property(doc='The path to the csv file, containing the raw data')
+    fs: float = Property(doc='Sampling frequency (Hz)')
+    omega: float = Property(doc='Signal frequency (Hz)')
+    sensor_loc: str = Property(doc='Cartesian coordinates of the sensors in the format\
```

Comment:
I think a `Sequence[StateVector]` would be better for both beamformers.

Reply:
Just trying to understand this - is the issue that I'm passing the csv file / data directly into the detector rather than using a reader and passing everything as a `Sequence[StateVector]`?

Reply:
It's the fact that your passing in sensor locations as a string of numbers, rather than just passing the numbers as an array, or list of arrays for example.

Reply:
Got it, thanks! This has been updated in the latest version.